assets/js/modules/analytics-4/datastore/properties.test.js

```diff
@@ -1308,6 +1381,186 @@ describe( 'modules/analytics-4 properties', () => {
+			} );
+
+			it( 'should convert property time to unix timestamp when value is set', () => {
+				const createTimeInDateString = '2014-10-02T15:01:23Z';
+
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.setPropertyCreateTime( createTimeInDateString );
+
+				const propertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( propertyCreateTime ).toBe(
+					new Date( createTimeInDateString ).getTime()
+				);
+			} );
+
+			it( 'should reset create time when new property is created', async () => {
+				const accountID = fixtures.createProperty.parent;
+
+				fetchMock.post( createPropertyEndpoint, {
+					body: fixtures.createProperty,
+					status: 200,
+				} );
+
+				fetchMock.get( propertyEndpoint, {
+					body: {},
+					status: 200,
+				} );
+
+				await registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.createProperty( accountID );
+
+				const propertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( propertyCreateTime ).toBeUndefined();
+			} );
+
+			it( 'should reset create time when property is changed', async () => {
+				const propertyID = '1000';
+
+				const settings = {
+					propertyID: '12345',
+					webDataStreamID: '1000',
+					measurementID: 'abcd',
+					propertyCreateTime: 1662715085968,
+				};
+
+				provideUserAuthentication( registry );
+
+				fetchMock.get( propertyEndpoint, {
+					body: {},
+					status: 200,
+				} );
+
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.receiveGetSettings( settings );
+
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.receiveGetWebDataStreams( fixtures.webDataStreams, {
+						propertyID,
+					} );
+
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.selectProperty( propertyID );
+
+				const propertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				await untilResolved(
+					registry,
+					MODULES_ANALYTICS_4
+				).getPropertyCreateTime();
+
+				expect( propertyCreateTime ).toBe( 0 );
+
+				expect( fetchMock ).toHaveFetched( propertyEndpoint, {
+					query: { propertyID },
+				} );
+			} );
+
+			it( 'should not invoke getProperty resolver if property is already received', () => {
+				const propertyID = fixtures.properties[ 0 ]._id;
+
+				const settings = {
+					propertyID,
+					webDataStreamID: '1000',
+					measurementID: 'abcd',
+				};
+
+				fetchMock.get( propertyEndpoint, {
+					body: {},
+					status: 200,
+				} );
+
+				provideUserAuthentication( registry );
+
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.receiveGetSettings( settings );
+
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.receiveGetProperty( fixtures.properties[ 0 ], {
+						propertyID,
+					} );
+
+				const propertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( propertyCreateTime ).toBeUndefined();
```

Comment:
As mentioned before, the setting should be `0` by default. Here we've explicitly received a partial set of settings which is why it is undefined.